lib/apollo-federation/schema.rb

```diff
@@ -53,20 +55,53 @@ def query(new_query_object = nil)
         federation_query_object
       end
 
+      def compose_directives
+        @compose_directives || find_inherited_value(:compose_directives, [])
+      end
+
+      def links
+        @links || find_inherited_value(:links, [])
+      end
+
+      def all_links
+        imported_directives = IMPORTED_DIRECTIVES
+        default_link = {
+          url: "https://specs.apollo.dev/federation/v#{federation_version}",
+          import: imported_directives,
+        }
+        default_link[:as] = default_link_namespace if default_link_namespace != DEFAULT_LINK_NAMESPACE
+        [default_link, *links]
+      end
+
       private
 
       def original_query
         @orig_query_object || find_inherited_value(:original_query)
       end
 
       def federation_2_prefix
-        federation_namespace = ", as: \"#{link_namespace}\"" if link_namespace != DEFAULT_LINK_NAMESPACE
+        schema = ['extend schema']
+
+        all_links.each do |link|
+          link_str = '  @link('
+          link_str += "url: \"#{link[:url]}\""
+          link_str += ", as: \"#{link[:as]}\"" if link[:as]
+          if link[:import]
+            imported_directives = link[:import].map { |d| "\"@#{d}\"" }.join(', ')
+            link_str += ", import: [#{imported_directives}]"
+          end
+          link_str += ')'
+          schema << link_str
+        end
```

Comment:
just pointing that this code annoys me but I dont have imporement suggestion :D
bu the spaces within the string are something xD

Reply:
That's the original code made in https://github.com/Gusto/apollo-federation-ruby/pull/253 without any changes from my side